angular.json

```diff
@@ -1,10 +1,16 @@
 {
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
```

Comment:
Are the changes for Angular CLI structure throughout this PR captured by a schematic? To what extend will these changes be required of other Angular CLI users? (Particularly ones who started a project in the early days like us)

cc @vikerman @IgorMinar @StephenFluin

Reply:
Most of the changes are not breaking or required changes. They mostly fall into the category of best practices. Over time, as these slight deviations from best practices build up, you can run into issues and/or unexpected breaking changes in future migrations. 

Most projects that I've seen will try to align themselves with the best practices (i.e. output from a `ng new`) from the latest version of the Angular CLI every couple of major releases. Currently, this repo seems to be aligned with the recommendations from Angular CLI v5.

Reply:
I guess what I really want to know here is how much of this change was for ivy and how much was was just event horizon for Angular CLI?

Reply:
I just went through and added comments for any Ivy-specific issues.